### PR TITLE
Async encoding of profiles to reduce latency impact

### PR DESCRIPTION
**Pull request by Qard**
I've added an `encodeAsync` method to `Profile` which splits up encoding into separate ticks by section of the profile data. This should help reduce latency impact of encoding.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1054,11 +1067,50 @@ export class Profile {
...
+    offset = this._encodeLocationsToBuffer(buffer, offset)
+    offset = this._encodeFunctionsToBuffer(buffer, offset)
+    offset = this.stringTable._encodeToBuffer(buffer, offset)
+    offset = this._encodeBasicValuesToBuffer(buffer, offset)
+    return offset
+  }
+
+  async _encodeToBufferAsync(buffer: Uint8Array, offset = 0): Promise<number> {
+    offset = this._encodeSampleTypesToBuffer(buffer, offset)
+    await new Promise(setImmediate)
```
Comment (szegedi) [inline src/index.ts:1083]: So _that's_ how you do `Thread.yield()` equivalent in Node! :-)

**Review by szegedi — APPROVED** [review]
I didn't expect you'll whip this up in few hours after we talked about it, but thank you!

File: src/index.ts (hunk 1 above)
Comment (Qard) [reply src/index.ts:1083]: The `setImmediate` could also be `queueMicrotask`, but it's less greedy this way, which is better for the latency-avoidance purpose. 😅

**Comment by Qard** [comment]
It was a simple idea I had in my head and was just as fast to implement as to explain, so I just went and wrote it. 😅

**Review by nsavoire — APPROVED** [review]

**Pull request merged**